src/engine/universal/core/src/messaging-setup.js

```diff
@@ -58,4 +59,68 @@ module.exports = {
       }
     }
   },
+  async setupMonitoringAndLogging(messaging, configModule, machineModule, logger) {
+    let { serverAddress, baseTopic } = await configModule.readConfig('messaging');
+    if (!serverAddress) return;
+
+    if (baseTopic && !baseTopic.endsWith('/')) baseTopic += '/';
+    baseTopic += 'proceed-pms';
+
+    // Monitoring data
+    const [{ id: machineId }, loadInterval] = await Promise.all([
+      machineModule.getMachineInformation(['id']),
+      configModule.readConfig('engine.loadInterval'),
+    ]);
+
+    setInterval(async () => {
+      try {
+        const machineData = await machineModule.getMachineInformation();
+        await messaging.publish(`${baseTopic}/engine/${machineId}/machine/monitoring`, machineData);
+      } catch (e) {
+        logger.error('Failed to publish monitoring data');
+      }
+    }, loadInterval * 1000);
+
+    // Logging data
+    const consoleLevel = await configModule.readConfig('logs.mqttLevel');
```

Comment:
Maybe rename this

Reply:
do you mean the `consoleLevel` of `logs.mqttLevel`?

Reply:
or*

Reply:
The `consoleLevel` because in the engines config `consoleLevel` refers to which messages should appear in the command line output